projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -328,6 +328,27 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
         return !!this.childLayoutKeys.length;
     }
 
+    /**
+     * @hidden
+     */
+    public get resolveRowEditContainer() {
+        return this.rootGrid ? this.rootGrid.rowEditContainer : this.rowEditContainer;
+    }
+
+    /**
+     * @hidden
+     */
+    public get resolveRowEditActions() {
+        return this.rootGrid ? this.rootGrid.rowEditActions : this.rowEditActions;
+    }
+
+    /**
+     * @hidden
+     */
+    public get resolveRowEditText() {
+        return this.rootGrid ? this.rootGrid.rowEditText : this.rowEditText;
```

Comment:
Shouldn't we allow setting different templates per rowIsland level? With the current implementation child grids will always use the template from the root grid so they will not take into account any template set on the row island level.

Reply:
I'm not sure since I'm not able to find that part in the spec. 
If that's the case however, it has to be decided whether a custom template in the parent grid and a lack of one in the child grids should mean that the child outlets should resolve to the parent's outlet (or to the default one). 
In case templating of the child grids' overlay should not be possible then this approach is cool, unless we decide to disable templates in the child grids at all, which makes no sense to me.

Reply:
IMO it should follow the same logic as the other templates:
https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts#L397

It should use the template defined in its own row island definition, otherwise use the default one.
Using the one from the parent doesn't make sense to me.

Reply:
The initial problem was that a row editing overlay was declared in a parent grid and was not shown in their successors. In my opinion, this is a legit use case scenario. If you don't think so, we can discuss it in the issue.
If you think that it's suitable, I can edit my fix to make a check if a template has been declared in the row island too, in order to allow customization of the overlay per children.

Reply:
I don't think that was the original issue. AFAIU the original issue states that all child layouts show the parent action strip instead of their own customized ones.

Reply:
Now I see, I'm sorry, I got the original issue wrong.